### ISSUE TITLE
[-] gets replaced with &shy; in the template editor

### ISSUE THREAD

**Issue by fritzmg** [contao/core-bundle#1075]
I just noticed that whenever you edit a template in the (ACE) editor in the back end, things like `[-]` automatically get replaced by their counterparts (i.e. `&shy;` in this case) when saving. However in my particular case I did not want that.

Should the code/template editor really replace those tags when saving, rather than having the tags replaced in the front end output?

**Comment by ausi** [comment]
I think I had as similar issue somewhere in the past while editing some PHP code in a template. Maybe we should access the raw value from the Symfony request in the template editor instead of relying on the `Input` class.

**Comment by leofeyer** [comment]
I don't think it has to do with the request. I assume the entities are replaced in the back end template?

**Comment by ausi** [comment]
It has to do with the `Input` and with `DC_Folder`. In https://github.com/contao/core-bundle/blob/60970bc1027a9b293279342ec0158df2b471da73/src/Resources/contao/drivers/DC_Folder.php#L1939 `Input::postRaw()` converts all `&shy;` to `[-]` and afterwards `StringUtil ::restoreBasicEntities()` converts `[-]` back to `&shy`. This causes issues if you want to save the string `[-]` somewhere in your Template because it gets converted to `&shy;`.

I think this can be solved by replacing the line from above with something like this:

```php
$strSource = System::getContainer()->get('request_stack')->getCurrentRequest()->request->get('source');
```

Similar to what we did in #708

**Comment by leofeyer** [comment]
Fixed in 3f933aa12ce0ee64e0036cd788763181952ce8f6.
